spec/datadog/tracing/contrib/grpc/distributed/propagation_spec.rb

```diff
@@ -0,0 +1,35 @@
+# typed: false
+
+require 'spec_helper'
+
+require 'datadog/tracing/contrib/grpc/distributed/propagation'
+require_relative '../../../distributed/propagation_spec'
+
+RSpec.describe Datadog::Tracing::Contrib::GRPC::Distributed::Propagation do
+  it_behaves_like 'Distributed tracing propagator' do
+    subject(:propagation) { described_class.new }
+
+    describe '.extract' do
+      subject(:extract) { propagation.extract(data) }
+      let(:trace_digest) { extract }
+
+      # Metadata values can also be arrays
+      # https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md
+      context 'given populated data in array format' do
+        let(:data) do
+          { 'x-datadog-trace-id' => %w[12345 67890],
+            'x-datadog-parent-id' => %w[98765 43210],
```

Comment:
This looks weird to me, as it somehow doesn't seem realistic. Other than a bug, would we expect this to ever happen in for these specific headers?

It may be worth commenting here (or in the original class) why we include support for this, as I myself can't really figure it out. Is it just future-proofing? In case we actually add a new header that uses arrays?

Reply:
gPRC metadata is a list of `key,value` tuples, and keys can be repeated: https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md#constructing-metadata

The Ruby gRPC gem "conveniently" uses a Hash to represent this data structure. When more than one value is present, the key becomes an array. I tested this and indeed it can be both an integral or list value.

There's a comment about this in the gPRC fetcher: `# Metadata values can be arrays (multiple values for the same key)`

Reply:
I expanded that comment in the Fetcher as well.

Reply:
Indeed, I can understand that in the general case, gprc metadata can be arrays.

But my question was more in the particular case of these keys -- are they expected in regular operation to ever be arrays?

E.g., is an array for any of these keys an indication that something is wrong, or something that is normal and happens every day? That's the part that puzzled me a bit.

(And if it's an error, should we perhaps log or mark the occurrence in some way?)

Reply:
Let's leave it as-is then, until new info on this shows up :)